R/numeric.R

```diff
@@ -42,15 +42,17 @@ vec_math <- function(fun, x, ...) {
 
 #' @export
 vec_math.default <- function(fun, x, ...) {
-  if (is_double(x)) {
-    vec_restore(vec_math_base(fun, x, ...), x)
-  } else if (fun %in% c("any", "all") && is_logical(x)) {
-    vec_restore(vec_math_base(fun, x, ...), x)
-  } else {
-    # nocov start
+  if (!is_double(x) && !(is_logical(x) && fun %in% c("any", "all"))) {
     stop_unimplemented(x, "vec_math")
-    # nocov end
   }
+
+  out <- vec_math_base(fun, x, ...)
+
+  if (!fun %in% c("any", "all", "is.finite", "is.infinite", "is.nan")) {
```

Comment:
What if we just restore if `out` is also double?

Reply:
This also works, done.